frontend/src/components/Ranking/Ranking.jsx

```diff
@@ -6,24 +6,37 @@ import Card from "react-bootstrap/Card";
 import Button from "react-bootstrap/Button";
 import { Context } from "../../context";
 
-const STOCKS_TO_SHOW = 10;
+const DEFAULT_RANKING_PAGE = 1;
+const MAX_PAGE_SIZE = 10;
 
 export default function Ranking() {
     const [stocksRanking, setStocksRanking] = useState([]);
-    const { setErrorMessage, setLoading } = useContext(Context);
-    const [stocksToShow, setStocksToShow] = useState(STOCKS_TO_SHOW);
+    const { setErrorMessage, loading, setLoading } = useContext(Context);
+    const [page, setPage] = useState(DEFAULT_RANKING_PAGE);
+    const [showLoadMore, setShowLoadMore] = useState(true);
 
     useEffect(() => {
         const fetchRanking = async () => {
             try {
                 setLoading(true);
                 const response = await axios.get(
-                    `${import.meta.env.VITE_HOST}/ranking`,
+                    `${import.meta.env.VITE_HOST}/ranking/${page}`,
                     { withCredentials: true, validateStatus: () => true }
                 );
 
                 if (response.status === 200) {
-                    setStocksRanking(response.data.stocksRanking);
+                    if (response.data.stocksRanking.length < MAX_PAGE_SIZE) {
+                        setShowLoadMore(false);
+                    }
+
+                    if (page === DEFAULT_RANKING_PAGE) {
```

Comment:
I think you don't need this check, `setStocksRanking` with `...prevRanking` should work on every case, because the initial state is an empty array.

This is still good working code:
```
[
  ...[],
  newItem
]
```